### ISSUE TITLE
DOC: more complete docs on offsets/frequencies needed

### ISSUE THREAD

**Issue by jorisvandenbossche** [pandas-dev/pandas#8438]
Add these to the API docs:

The contents of `pd.tseries.offsets.__all__`

```
['Day',
 'BusinessDay',
 'BDay',
 'CustomBusinessDay',
 'CDay',
 'CBMonthEnd',
 'CBMonthBegin',
 'MonthBegin',
 'BMonthBegin',
 'MonthEnd',
 'BMonthEnd',
 'SemiMonthEnd',
 'SemiMonthBegin',
 'BusinessHour',
 'CustomBusinessHour',
 'YearBegin',
 'BYearBegin',
 'YearEnd',
 'BYearEnd',
 'QuarterBegin',
 'BQuarterBegin',
 'QuarterEnd',
 'BQuarterEnd',
 'LastWeekOfMonth',
 'FY5253Quarter',
 'FY5253',
 'Week',
 'WeekOfMonth',
 'Easter',
 'Hour',
 'Minute',
 'Second',
 'Milli',
 'Micro',
 'Nano',
 'DateOffset']
```

and `pd.tseries.api.infer_freq`

---

From mailingslist (https://groups.google.com/forum/?fromgroups#!topic/pydata/66lTrtgev7A), eg `frequencies.is_super/subperiod` are missing in the docs (both in user guide as in api docs)

A lot of the functions/objects in `pandas.tseries.offsets` and `pandas.tseries.frequencies` are not included in the API docs, or mentioned in the timeseries.rst docs. 
Furthermore, the difference between both submodules is not fully clear to me (eg the offsets objects like `BDay` are available in both)

**Comment by TomAugspurger** [comment]
I think most of the public API has been decided.

The TODO here is to add the contents of `pd.tseries.offsets.__all__`

```
['Day',
 'BusinessDay',
 'BDay',
 'CustomBusinessDay',
 'CDay',
 'CBMonthEnd',
 'CBMonthBegin',
 'MonthBegin',
 'BMonthBegin',
 'MonthEnd',
 'BMonthEnd',
 'SemiMonthEnd',
 'SemiMonthBegin',
 'BusinessHour',
 'CustomBusinessHour',
 'YearBegin',
 'BYearBegin',
 'YearEnd',
 'BYearEnd',
 'QuarterBegin',
 'BQuarterBegin',
 'QuarterEnd',
 'BQuarterEnd',
 'LastWeekOfMonth',
 'FY5253Quarter',
 'FY5253',
 'Week',
 'WeekOfMonth',
 'Easter',
 'Hour',
 'Minute',
 'Second',
 'Milli',
 'Micro',
 'Nano',
 'DateOffset']
```

and `pd.tseries.api.infer_freq`

**Comment by mroeschke** [comment]
Closed by #23405
